spec/parsers/pdf_parser_spec.rb

```diff
@@ -1,49 +1,71 @@
         'rb'
       )
     )
-  }
-
-  shared_examples :behave_like_pdf do |hash|
-    let(:pdf_file) { hash.fetch(:file) }
+  end
 
-    it 'acts as a pdf' do
+  shared_examples :behave_like_pdf do |params|
+    it "#{params[:file]} acts as a pdf" do
+      parsed_pdf = parse_pdf params[:file]
       expect(parsed_pdf).not_to be_nil
       expect(parsed_pdf.nature).to eq(:document)
       expect(parsed_pdf.format).to eq(:pdf)
       expect(parsed_pdf.content_type).to eq('application/pdf')
     end
   end
 
-  describe 'a PDF file with a missing version header' do
-    let(:pdf_file) { 'not_a.pdf' }
-
-    it 'does not parse succesfully' do
-      expect(parsed_pdf).to be_nil
+  describe 'parses a PDF file' do
+    describe 'a single page file' do
+      include_examples :behave_like_pdf, file: '1_page.pdf'
     end
-  end
 
-  describe 'a PDF file with a correct header but no valid content' do
-    let(:pdf_file) { 'broken.pdf' }
+    describe 'various PDF versions' do
+      include_examples :behave_like_pdf, file: 'Lorem Ipsum PDF 1.6.pdf'
+      include_examples :behave_like_pdf, file: 'Lorem Ipsum PDF-A-1b.pdf'
+      include_examples :behave_like_pdf, file: 'Lorem Ipsum PDF-A-2b.pdf'
+      include_examples :behave_like_pdf, file: 'Lorem Ipsum PDF-A-3b.pdf'
+      include_examples :behave_like_pdf, file: 'Lorem Ipsum PDF-UA.pdf'
+      include_examples :behave_like_pdf, file: 'Lorem Ipsum Hybrid - ODF embedded.pdf'
+      include_examples :behave_like_pdf, file: 'Simple PDF 2.0 file.pdf'
+    end
 
-    pending 'does not parse succesfully'
+    describe 'complex PDF 2.0 files' do
+      include_examples :behave_like_pdf, file: 'PDF 2.0 image with BPC.pdf'
+      include_examples :behave_like_pdf, file: 'PDF 2.0 UTF-8 string and annotation.pdf'
+      include_examples :behave_like_pdf, file: 'PDF 2.0 via incremental save.pdf'
+      include_examples :behave_like_pdf, file: 'PDF 2.0 with page level output intent.pdf'
+      include_examples :behave_like_pdf, file: 'pdf20-utf8-test.pdf'
+    end
   end
 
-  describe 'exceeding the PDF read limit' do
-    let(:pdf_file) { 'read_limit.pdf' }
+  describe 'broken PDF files should not parse' do
+    it 'PDF with missing version header' do
+      parsed_pdf = parse_pdf 'not_a.pdf'
+      expect(parsed_pdf).to be_nil
+    end
 
-    pending 'does not parse succesfully'
-  end
+    it 'PDF 2.0 with offset start' do
+      parsed_pdf = parse_pdf 'PDF 2.0 with offset start.pdf'
+      expect(parsed_pdf).to be_nil
+    end
 
-  describe 'parses a PDF file' do
-    describe 'a single page file' do
-      include_examples :behave_like_pdf, file: '1_page.pdf'
+    describe 'PDF file with a correct header but no valid content' do
+      # parsed_pdf = parse_pdf 'broken.pdf'
+      pending 'does not parse succesfully'
+    end
+
+    describe 'when exceeding the PDF read limit' do
+      # parsed_pdf = parse_pdf 'read_limit.pdf'
+      pending 'does not parse succesfully'
```

Comment:
Do pending tests do anything? I'd just remove them unless they are testing something 🤷

Reply:
Realistically, are we ever coming back to implement this?
It feels like a `# todo` in the code. I'd be fine either removing it altogether or implementing this.

Reply:
I've removed them.

Reply:
...or that. I think it would be good creating an issue or something to come back to this. I definitely see value in having these tests properly implemented.

Reply:
@linkyndy Yes, this is how they are in the 'master' branch:
<img width="559" alt="image" src="https://user-images.githubusercontent.com/2901534/227144226-37637934-9742-4bdd-b661-8d25f0de0e93.png">

Reply:
@linkyndy Looking at the GIT history for those tests, I see they were created like this in 2018, with the commit message "Rewritten the PDF parser".

Reply:
Ok I've implemented the `exceeds the PDF read limit` test. Check it out.
I'm not implementing the invalid content test, because that requires implementing a full PDF parser which I feel is out of scope for this PR.

Reply:
Cool 👍 

I'm okay with not implementing that test in this PR, but let's at least keep a trace that we should do that at some point. Either works, leave the pending test or create a Jira issue.

Reply:
Added Jira ticket.